Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -409,7 +409,7 @@ public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomString
         try container.encode(url.absoluteString, forKey: .url)
         
         let sourceLanguageIDVariants = DocumentationDataVariants<String>(
```

Comment:
Found a few other places where we could use modern syntax for dictionaries:

DocumentationContext.swift
```
  
    /// The documentation bundles that are currently registered with the context.
    public var registeredBundles: Dictionary<String, DocumentationBundle>.Values {
        return dataProvider.bundles.values
    }
```

BidirectionalMap.swift
```
 func makeIterator() -> Dictionary<Value1, Value2>.Iterator {
        return forward.makeIterator()
    }
```

GroupedSequence.swift
```
    /// Returns an iterator over the members of the sequence.
    func makeIterator() -> Dictionary<Key, Element>.Values.Iterator {
        storage.values.makeIterator()
    }
```

Reply:
Some of these can be written using opaque types (the `some` keyword) to not define a specific type. For example:
```diff
- public var registeredBundles: Dictionary<String, DocumentationBundle>.Values 
+ public var registeredBundles: some Collection<DocumentationBundle>
```
```diff 
  /// Returns an iterator over the members of the sequence.
- func makeIterator() -> Dictionary<Key, Element>.Values.Iterator {
+ func makeIterator() -> some IteratorProtocol<Element> { 
```

but for the bidirectional map I don't feel like `some IteratorProtocol<(key: Value1, value: Value2)>` is any more readable than `[Value1: Value2].Iterator`

Reply:
I agree - `some IteratorProtocol<Element>` is actually quite readable even though it uses the verbose generic syntax. Calling this will return some something that can iterate over Element's.

But yea `some IteratorProtocol<(key: Value1, value: Value2)>` seems confusing either because there's a key and a value in the tuple, or just because of the unhelpful names `Value1` and `Value2`.